src/main/java/com/linkedin/kmf/services/ClusterTopicManipulationService.java

```diff
@@ -171,7 +175,37 @@ private boolean doesClusterContainTopic(String topic, Collection<Node> brokers,
 
       totalPartitionsInCluster += this.processBroker(logDirectoriesResponseMap, broker, topic);
     }
-    return totalPartitionsInCluster == expected;
+    return totalPartitionsInCluster == expectedTotalPartitionsInCluster
```

Comment:
Actually what I meant is:

```
if (totalPartitionInCluster != expected) {
    return false;
}

boolean describeSucceed = true;
try {
    Map<String, TopicDescription> topicDescriptions = ClusterTopicManipulationService.describeTopics(adminClient, topicNames);
    LOGGER.trace("topicDescriptionMap = {}", topicDescriptions);
} catch (InterruptedException | ExecutionException e) {
    describeSucceed = false;
    LOGGER.error("Exception occurred within describeTopicsFinished method for topics {}", topicNames, e);
}

return !describeSucceed;
```

In the above approach

1. We removed the `isTopicDescribeSuccessful` method. In your updated version, `isTopicDescribeSuccessful` is still doing two things, waiting and return boolean. You just factored out the waiting part to its own function. But that does not change the semantics to `isTopicDescribeSuccessful`

2. The code is very readable since readers would know that we wait and if results are returned successfully, we are good to go. Instead of looking at the `isTopicDescribeSuccessful` method, reading its comments and then realizing that this function blocks for results and log the exception if there is any, etc.

3. Reduce the number of static functions we have. There are some Java developers (I am not one of them) who are very against the idea of having static method in a class unless a class is used as a pure utility class. But I do agree that we should try to avoid unnecessary static functions.

Reply:
thanks for your opinion